llvm/lib/IR/SSAContext.cpp

```diff
@@ -68,6 +69,23 @@ bool SSAContext::isConstantOrUndefValuePhi(const Instruction &Instr) {
   return false;
 }
 
+template <>
+void SSAContext::getPhiInputs(
+    const Instruction &Instr, SmallVectorImpl<const Value *> &Values,
+    SmallVectorImpl<const BasicBlock *> &Blocks) const {
+  assert(isa<PHINode>(Instr));
+  const PHINode *Phi = static_cast<const PHINode *>(&Instr);
+  for (unsigned I = 0, E = Phi->getNumIncomingValues(); I != E; ++I) {
+    const Value *Incoming = Phi->getIncomingValue(I);
+    const BasicBlock *Block = Phi->getIncomingBlock(I);
+    // FIXME: should this also consider Incoming == &Instr undef?
```

Comment:
If all the dominating inputs are well-defined, then this self-reference is not undef, right? If at this point we find that indeed `Incoming == &Instr`, then I am not able to decide if treating it as `undef` is always correct. Ultimately, it the uniformity itself has to be `false` unless proven otherwise.